lib/hydra/derivatives.rb

```diff
@@ -118,7 +118,7 @@ def constantize_processor(processor)
     end
 
     def transform_datastream(file_name, transform_directives, opts={})
```

Comment:
So ... what do you do if opts is empty, since the default image processor
will cause a failure.  Should an error be raised if opts is empty?

if opts.empty?
  raise 'Options must indicate processor to use'
end

On Fri, Aug 7, 2015 at 4:25 AM, stkenny <notifications@github.com> wrote:

> In lib/hydra/derivatives.rb
> <https://github.com/projecthydra/hydra-derivatives/pull/69#discussion_r36499602>
> :
>
> > @@ -118,7 +118,7 @@ def constantize_processor(processor)
> >      end
> >
> >      def transform_datastream(file_name, transform_directives, opts={})
>
> Hi, the problem was that if the opts are empty then the only processor
> that can be used is the image processor. The processor to use is picked up
> from the options hash. If the options are empty the default is :image.
>
>  def initialize_processor(file_name, transform_directives, opts={})
>       processor_class(opts[:processor] || :image).new(self, file_name, transform_directives, opts)
> end
>
> If this spec, spec/units/transcoding_spec.rb, was changed to use
> transform_datastream instead of transform_file it would likely fail on
> anything other than the image processing tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projecthydra/hydra-derivatives/pull/69/files#r36499602>
> .
>